app/src/store/playground/playgroundStore.tsx

```diff
@@ -290,11 +288,55 @@ export const createPlaygroundStore = (
           }
         }
       });
-      set({ input: { variables: { ...variables } } });
+      set({
+        input: { ...get().input, variableKeys: [...Array.from(variables)] },
+      });
+    },
+    setVariableValue: (key: string, value: string) => {
+      const input = get().input;
+      if ("variablesValueCache" in input) {
```

Comment:
It might be worth it to just split the "input" field into two fields; One for manualInput and datasetInput so that we don't have to do all of this type narrowing in the store.

I suppose that means we'd have to do it in our selectors or components instead but it depends on how much will change in the dataset version of the playground